Library/Homebrew/dev-cmd/create.rb

```diff
@@ -23,6 +23,8 @@ def create_args
              description: "Create a basic template for an Autotools-style build."
       switch "--cmake",
              description: "Create a basic template for a CMake-style build."
+      switch "--go-mod",
+             description: "Create a basic template for a Go-Modules-style build."
```

Comment:
What about just making this `a Go build.`?

Reply:
I think it would be too ambigous. Go currently has 2 official build "styles". One newer with Modules without GOPATH and one older with GOPATH.